src/test/java/org/humancellatlas/ingest/AutoLoaderTest.java

```diff
@@ -0,0 +1,163 @@
+package org.humancellatlas.ingest;
+
+import org.humancellatlas.ingest.client.IngestApiClient;
+import org.humancellatlas.ingest.model.SubmissionEnvelopeReference;
+import org.humancellatlas.ingest.state.*;
+import org.humancellatlas.ingest.state.monitor.SubmissionStateMonitor;
+import org.humancellatlas.ingest.state.persistence.AutoLoader;
+import org.humancellatlas.ingest.state.persistence.AutoLoaderFailureException;
+import org.humancellatlas.ingest.state.persistence.Persister;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.statemachine.StateMachine;
+import org.springframework.statemachine.config.StateMachineFactory;
+import org.springframework.statemachine.data.StateMachineRepository;
+import org.springframework.statemachine.data.redis.RedisRepositoryStateMachine;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestClientException;
+
+import java.net.URI;
+import java.util.*;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {AutoLoader.class, StateMachineConfiguration.class })
+public class AutoLoaderTest {
```

Comment:
This test does a lot of setup and mocking in order to test the state machine startup logic. It makes it more difficult to read and maintain.

Reply:
I'm afraid I don't know how else to test them without mocking those services/components I do not control.

Reply:
Usually many inits means a class that does too many things with too many dependencies. I think we can let this one go, and pay more attention in the future.